### DISCUSSION TITLE
array field 设置初始值的时候没有触发设置的联动

### DISCUSSION THREAD

**Discussion by fushizhe** [alibaba/formily#2364] [Q&A 2.x]
**codesandbox**:   https://codesandbox.io/s/gifted-cherry-x52z2?file=/App.tsx

  求教个array field的联动问题哈。
  schema是一个array field ，每个数组对象包括一个input1和input2，设置了联动，让input1值变化的时候同时同步到input2， settimeout模拟请求后端接口5S后给表单值初始化，期望是input1填入值test后能触发联动使input2也填入test，但是实际没有触发。

11.1补充：因为这个现象的存在，实际业务使用中在获取后端数据后得先往表单里设置一组和数组长度等长的空对象，然后再设置一次真实的数组值，才能触发每个数组对象内部的联动。不知道是不是我使用方式的问题，或者这种场景有没有推荐的最佳实践呢？

**Comment by janryWang** [comment]
你的例子里并没有settimeout逻辑呀

**Reply by fushizhe** [reply]
在最下面
![image](https://user-images.githubusercontent.com/2124201/139630136-88a04957-94ad-4d37-b402-264ece5b8b58.png)

**Reply by janryWang** [reply]
你这个场景，最佳实践不是这种主动模式，而是使用x-reactions被动联动，当然，你走主动模式的话，onFieldValueChange这个只有在存在默认值的时候，第一次才会触发，否则是不会触发的，当然，如果你硬是希望在useFormEffects里处理的话，是可以使用onFieldChange('xxx',['value','initialized'],()=>{})这样来组合状态监听，

**Reply by janryWang** [reply]
onFieldReact也是可以的
